stl/inc/iterator

```diff
@@ -135,15 +141,26 @@ public:
 
     constexpr istream_iterator() : _Myistr(nullptr), _Myval() {}
 
-    istream_iterator(istream_type& _Istr) : _Myistr(_STD addressof(_Istr)) {
+    istream_iterator(istream_type& _Istr)
+#if _HAS_CXX20
+        : _Myistr(_STD addressof(_Istr)), _Myval() {
+#else // ^^^ _HAS_CXX20 // !_HAS_CXX20 vvv
+        : _Myistr(_STD addressof(_Istr)) {
+#endif // _HAS_CXX20
         _Getval();
     }
 
     _NODISCARD const _Ty& operator*() const {
+#if _ITERATOR_DEBUG_LEVEL > 0
+        _STL_VERIFY(_Myistr, "in_stream must be non null");

```

Comment:
[`_STL_ASSERT`](https://github.com/microsoft/STL/blob/master/stl/inc/yvals.h#L197-L201) is already a conditional `_STL_VERIFY`; we can use it here and avoid the conditional compilation. (Also on 162.)

Reply:
I know `in_stream` is the name the Standard uses for `_Myistr`, but the reader of this error message probably won't. Maybe call it "The stored stream pointer"? (And lines 162 and 185.)

Reply:
_STL_ASSERT is controlled by _DEBUG/NDEBUG, not _ITERATOR_DEBUG_LEVEL. If it's an IDL based thing it should use _STL_VERIFY guarded, because we have customers who want to turn on IDL support in release builds.

Reply:
That said this particular assert doesn't look like it should be affected by IDL. Yes this is a debug check, and yes this is an iterator, but IDL refers specifically to the containers and iterators can track each other machinery....

Reply:
`NDEBUG` controls the `assert` macro; it is not related to `_DEBUG` being defined.